cou/utils/app_utils.py

```diff
@@ -44,9 +44,59 @@ async def upgrade_packages(units: Iterable[str], model: COUModel) -> None:
             if str(result["Code"]) == "0":
                 logger.debug(result["Stdout"])
             else:
-                raise PackageUpgradeError(
-                    f"Cannot upgrade packages on {unit}."
-                ) from CommandRunFailed(cmd=command, result=result)
+                raise RunUpgradeError(f"Cannot upgrade packages on {unit}.") from CommandRunFailed(
+                    cmd=command, result=result
+                )
 
         except JujuError as exc:
-            raise PackageUpgradeError(f"Cannot upgrade packages on {unit}.") from exc
+            raise RunUpgradeError(f"Cannot upgrade packages on {unit}.") from exc
+
+
+async def set_require_osd_release_option(unit: str, model: COUModel, ceph_release: str) -> None:
```

Comment:
if this function is specific to the ceph-mon class, why put it here?

Reply:
Because ideally in the future, I think we should have all upgrade related functions put in a single file, like the current `app_utils.py`. And the application class file should just contain the knowledge about which steps an application needs etc.